src/pyJiraCli/cmd_delete.py

```diff
@@ -0,0 +1,164 @@
+    """
+
+    sb_login = subparser.add_parser('delete',
+                                    help="delete login information")
+
+    # pylint: disable=duplicate-code
+
+    option_grp = sb_login.add_argument_group('data type to delete')
+
+    option_grp.add_argument('--default',
+                            '-d',
+                            action='store_true',
+                            help="primary server url to use")
+
+    option_grp.add_argument('--userinfo',
+                            '-ui',
+                            action='store_true',
+                            help="username, pw")
+
+    option_grp.add_argument('--server',
+                            '-s',
+                            action='store_true',
+                            help="second server url")
+
+    option_grp.add_argument('--token',
+                            '-t',
+                            action='store_true',
+                            help="API token for jira server")
+
+    option_grp.add_argument('--cert',
+                            '-c',
+                            action='store_true',
+                            help="authentification certificate for jira server")
+
+    # pylint: disable=duplicate-code
+
+    return sb_login
+
+def execute(args) -> Ret:
+    """ This function servers as entry point for the command 'delete'.
+        It will be stored as callback for this moduls subparser command.
+    
+    Args: 
+        args (obj): The command line arguments.
+        
+    Returns:
+        Ret:   Returns Ret.RET_OK if succesfull or the corresponding error code if not.
+    """
+    return _cmd_delete(args)
+
+def _cmd_delete(args) -> Ret:
+    """ Delete login information.
+        
+    Args:
+        args (obj): The command line arguments.
+        
+    Returns:
+        Ret:   Ret.RET_OK if succesfull, corresponding error code if not
+    """
+    ret_status =  _delete_login_file(args.userinfo,
+                                         args.token,
+                                         args.server,
+                                         args.default,
+                                         args.cert)
+    return ret_status
+
+def _delete_login_file(delete_userinfo:bool,
+                       delete_token:bool,
+                       delete_server:bool,
+                       delete_default_server:bool,
+                       delete_certificate:bool) -> Ret:
+    """ Delete the login files corresponding to the set dataType flags.
+
+    Args:
+        delete_userinfo (bool):         Flag to delete userinfo data.
+        delete_token (bool):            Flag to delete token data.
+        delete_server (bool):           Flag to delete server data.
+        delete_default_server (bool):   Flag to delete default server.
+        delete_certificate (bool):      Flag to delete the server certificate.
+
+    Returns:
+        Ret:   Returns Ret.RET_OK if succesfull or the corresponding error code if not.
```

Comment:
Two "if" ... "if not" ... use one to express it. "successfull" with only one "l" at the end.